File: homework/Nov23/miniProject/Page/DeleteProductPage.java
package lab.Nov26.miniProject.Page;

import java.util.Scanner;

import lab.Nov26.miniProject.Product;
import lab.Nov26.miniProject.ProductManager;
import lab.Nov26.miniProject.State;
import lab.Nov26.miniProject.Abstract.Page;

public class DeleteProductPage extends Page {

	public DeleteProductPage() {}

	@Override
	public void show() {
		ProductManager products = State.getState().getProductManager();
		products.showProducts();
		System.out.println("Type product Id to delete\nb. back");
		String input = State.getState().getScanner().next();
		switch (input) {
			case "b": {
				Page previousPage = (Page) State.getState().getStack().pop();
				previousPage.show();
				break;
			}
			default: {
				if (isNumber(input)) {
					int id = Integer.parseInt(input);
					Product p = products.getProductById(id);
					if (products.isContain(p)) {
						products.deleteProduct(p);
					} else {
						System.out.println("Id not found, Please try again");
					}
					show();
				}
				show();
				break;
			}
		}
	}
}

File: src/lab/miniProject_TS/src/page/deleteProductPage.ts
import { Page } from "../interface"
import Product from "../product"
import ProductManager from "../productManager"
import State from "../state"

class DeleteProductPage implements Page {
    constructor() {}

    public show(): void {
        const products: ProductManager = State.getState().getProductManager()
        products.showProducts()
        const rl = State.getState().getRl()
        rl.question('Type product Id to delete\nb. back\n', (input: string) => {
            switch (input) {
                case 'b': {
                    State.getState().getStack().pop()?.show()
                    break
                }
                default: {
                    if(!isNaN(+input)) {
                        let id: number = +input
                        let p: Product = products.getProduct(id)
                        if (products.isContain(p)) products.deleteProduct(p)
                        else console.log('Id not found, Please try again')
                    }
                    this.show()
                    break
                }
            }
        })
    }
}

export default DeleteProductPage
